### PR TITLE
AP_HAL_SITL: correct battery setup for rover

### PR DESCRIPTION
**Pull request by khancyr**
fix the battery consumtion for rover

### CODE REVIEW

**Comment by peterbarker** [comment]
Would Planes with reverse thrust also see this problem?

**Comment by khancyr** [comment]
if they are centered at 1500 yes ! 
This was just a temp patch ... I think that each SITL instance should update its own battery consumption, as here it fix only normal rover but not skid one ...

**Comment by rmackay9** [comment]
@peterbarker, you're happy for this to go in?  i'm not really an expert on SITL but it looks OK to me so if you're fine, and @khancyr is fine then let's merge it?

File: libraries/AP_HAL_SITL/SITL_State.cpp [hunk 1]
```diff
@@ -469,7 +469,12 @@ void SITL_State::_simulator_servos(struct sitl_input &input)
                 }
             } else {
                 // simulate simple battery setup
-                float throttle = motors_on?(input.servos[2]-1000) / 1000.0f:0;
+                float throttle;
+                if (_vehicle == APMrover2) {
```
Comment (peterbarker) [inline libraries/AP_HAL_SITL/SITL_State.cpp]: Isn't this also relevant for sub?

And Planes that support reverse throttle? (which I don't think we have a simulator for...)

Still, not after perfection.

**Comment by rmackay9** [comment]
Merged, hopefully @khancyr is happy to address the points raised by Peter.

**Pull request merged**